### PR TITLE
Web branch

### PR DESCRIPTION
**Pull request by DenisBogusevich**

### CODE REVIEW

File: src/main/java/onlinebookstore/service/impl/BookServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    @Override
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                        .map(bookMapper::toDto)
+                        .toList();
+    }
+
+    @Override
+    public BookDto getBookById(Long id) {
```
Comment (pavlov-n) [inline src/main/java/onlinebookstore/service/impl/BookServiceImpl.java]: I'm not sure, but maybe you need to name it findById(), cause of throwing exception

File: src/main/java/onlinebookstore/service/impl/BookServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        return bookRepository.findById(id)
+                            .map(bookMapper::toDto)
+                            .orElseThrow(() -> new EntityNotFoundException("Can`t find"
+                                    + "book with id:" + id));
```
Comment (pavlov-n) [inline src/main/java/onlinebookstore/service/impl/BookServiceImpl.java]: You can use here .formated()

File: src/main/java/onlinebookstore/service/impl/BookServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class BookServiceImpl implements BookService {
+    private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
+
+    @Override
+    public Book save(CreateBookRequestDto book) {
```
Comment (pavlov-n) [inline src/main/java/onlinebookstore/service/impl/BookServiceImpl.java]: As i know you need to return BookDto, not Book

File: src/main/java/onlinebookstore/dto/BookDto.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+    private String author;
+
+    private String isbn;
+
+    private BigDecimal price;
+
+    private String description;
+
+    private String coverImage;
```
Comment (pavlov-n) [inline src/main/java/onlinebookstore/dto/BookDto.java]: i think it should be like this:
```suggestion
    private String title;
    private String author;
    private String isbn;
    private BigDecimal price;
    private String description;
    private String coverImage;
```

File: src/main/java/onlinebookstore/controler/BookControler.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        return bookService.findAll();
+    }
+
+    @GetMapping("/{id}")
+    public BookDto getBookById(@PathVariable Long id) {
+        return bookService.getBookById(id);
+    }
+
+    @PostMapping
+    public Book createBook(@RequestBody CreateBookRequestDto bookDto) {
```
Comment (pavlov-n) [inline src/main/java/onlinebookstore/controler/BookControler.java]: May be you should return BookDto here

**Review by pavlov-n — APPROVED** [review]
Good job, i left some minor comments <3

File: src/main/java/onlinebookstore/dto/BookDto.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
+package onlinebookstore.dto;
+
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Data
+public class BookDto {
```
Comment (liliia-ponomarenko) [inline src/main/java/onlinebookstore/dto/BookDto.java]: Let’s  use records for dtos

File: src/main/java/onlinebookstore/repository/impl/BookRepositoryImpl.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+import java.util.Optional;
+import onlinebookstore.entity.Book;
+import onlinebookstore.exception.DataProcessingException;
+import onlinebookstore.repository.BookRepository;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
```
Comment (liliia-ponomarenko) [inline src/main/java/onlinebookstore/repository/impl/BookRepositoryImpl.java:16]: you can use RequiredArgsConstructor

File: src/main/java/onlinebookstore/repository/impl/BookRepositoryImpl.java [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+            throw new EntityNotFoundException("Can't get all books from DB", e);
+        }
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try {
+            return Optional.ofNullable(sessionFactory
+                    .fromSession(s -> s.find(Book.class, id)));
+        } catch (Exception e) {
```
Comment (liliia-ponomarenko) [inline src/main/java/onlinebookstore/repository/impl/BookRepositoryImpl.java]: Can you use try-with-resources?

File: pom.xml [hunk 9]
```diff
@@ -48,10 +58,63 @@
...
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-actuator</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>com.h2database</groupId>
+            <artifactId>h2</artifactId>
+        </dependency>
```
Comment (liliia-ponomarenko) [inline pom.xml:72]: why do you duplicate this dependency?

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Good job! Let’s fix a few mistakes ;)

File: src/main/java/onlinebookstore/dto/BookDto.java [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+
+public record BookDto(
+        String title,
+        String author,
+        String isbn,
+        BigDecimal price,
+        String description,
+        String coverImage) {
+}
+
```
Comment (liliia-ponomarenko) [inline src/main/java/onlinebookstore/dto/BookDto.java:13]:
```suggestion
```

**Review by liliia-ponomarenko — APPROVED** [review]
Good job!

**Pull request merged**